src/Engine/Renderer/RenderTechnique/RenderTechnique.hpp

```diff
@@ -13,79 +14,165 @@ namespace Ra {
 namespace Engine {
 class ShaderProgram;
 class Material;
+class ShaderParameterProvider;
 } // namespace Engine
 } // namespace Ra
 
 namespace Ra {
 namespace Engine {
-/* Radium V2 : make this class  non final and extensible.
- * passName must be renderer independant (use a map so that each renderer could set its own passes.
- * Rely this to future composition based material and renderer architecture.
+/**
+ * Default render pass ids
+ */
+namespace DefaultRenderingPasses {
+
+/**
+ * The prepass of Radium default renderer.
+ * This pass fill in the depth buffer and initialize feature buffers
+ */
+const Core::Utils::Index Z_PREPASS{1};
+/**
+ * The main pass of Radium default renderer.
+ * This pass render, with lighting, the fully opaque fragments
  */
+const Core::Utils::Index LIGHTING_OPAQUE{0};
+
 /**
- * Set of shaders to be used by the renderer to render objects with materials.
+ * The transpaency pass of Radium default renderer.
+ * This pass render, with lighting, the transparent fragments using LIT-OIT algorithm
+ * @see Ra::Engine::ForwardRenderer documentation
+ */
+const Core::Utils::Index LIGHTING_TRANSPARENT{2};
+/** DEFAULT_PASS will be used to draw debug or ui objects.
+ * This pass is expeted to draw the object with a computed color on fully opaque fragments
+ */
+const Core::Utils::Index DEFAULT_PASS{LIGHTING_OPAQUE};
+} // namespace DefaultRenderingPasses
```

Comment:
Why not using an enum here ?

Reply:
Because an enum could not be extended nor overriden. Using several constants (typed as an Index) allows a renderer to define as much rendering passes as he want with its proper numbering and semantic. An enum was used in old Radium and strongly limits the extensibility of the renderers and usability of the renderTechniques.